#<Redis::CannotConnectError: Error connecting to Redis on 0.0.0.0:6379 (Errno::ECONNREFUSED)>
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:398:in `rescue in establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:379:in `establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:115:in `block in connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:344:in `with_reconnect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:114:in `connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:417:in `ensure_connected'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:269:in `block in process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:356:in `logging'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:268:in `process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:161:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:263:in `block in send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `synchronize'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/commands/sets.rb:115:in `smembers'
/home/emr/apps/gxt/apps/esm-miot-monitor/service.rb:1272:in `block (2 levels) in registered'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/em/timers.rb:56:in `fire'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run_machine'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/backends/base.rb:75:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/server.rb:162:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/rack/handler/thin.rb:22:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1575:in `start_server'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1509:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-contrib-2.2.0/lib/sinatra/reloader.rb:260:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/main.rb:45:in `block in <module:Sinatra>'
#<Redis::CannotConnectError: Error connecting to Redis on 0.0.0.0:6379 (Errno::ECONNREFUSED)>
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:398:in `rescue in establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:379:in `establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:115:in `block in connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:344:in `with_reconnect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:114:in `connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:417:in `ensure_connected'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:269:in `block in process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:356:in `logging'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:268:in `process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:161:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:263:in `block in send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `synchronize'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/commands/sets.rb:115:in `smembers'
/home/emr/apps/gxt/apps/esm-miot-monitor/service.rb:1272:in `block (2 levels) in registered'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/em/timers.rb:56:in `fire'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run_machine'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/backends/base.rb:75:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/server.rb:162:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/rack/handler/thin.rb:22:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1575:in `start_server'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1509:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-contrib-2.2.0/lib/sinatra/reloader.rb:260:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/main.rb:45:in `block in <module:Sinatra>'
#<Redis::CannotConnectError: Error connecting to Redis on 0.0.0.0:6379 (Errno::ECONNREFUSED)>
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:398:in `rescue in establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:379:in `establish_connection'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:115:in `block in connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:344:in `with_reconnect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:114:in `connect'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:417:in `ensure_connected'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:269:in `block in process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:356:in `logging'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:268:in `process'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/client.rb:161:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:263:in `block in send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `synchronize'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis.rb:262:in `send_command'
/usr/local/lib/ruby/gems/3.1.0/gems/redis-4.6.0/lib/redis/commands/sets.rb:115:in `smembers'
/home/emr/apps/gxt/apps/esm-miot-monitor/service.rb:1272:in `block (2 levels) in registered'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/em/timers.rb:56:in `fire'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `call'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run_machine'
/usr/local/lib/ruby/gems/3.1.0/gems/eventmachine-1.2.7/lib/eventmachine.rb:195:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/backends/base.rb:75:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/thin/server.rb:162:in `start'
/usr/local/lib/ruby/gems/3.1.0/gems/thin-1.8.1/lib/rack/handler/thin.rb:22:in `run'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1575:in `start_server'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/base.rb:1509:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-contrib-2.2.0/lib/sinatra/reloader.rb:260:in `run!'
/usr/local/lib/ruby/gems/3.1.0/gems/sinatra-2.2.0/lib/sinatra/main.rb:45:in `block in <module:Sinatra>'
